openwisp_controller/connection/tests/test_models.py

```diff
@@ -274,14 +276,36 @@ def test_device_config_update(self):
                 }
             ]
         }
-        # here you can start capturing standard output
-        # see how it's done here: https://github.com/ncouture/MockSSH/blob/master/tests/test_mock_cisco.py#L14-L20
         c.full_clean()
         c.save()
-        # at this point the update_config() method will be triggered
-        # you need to create a command in the mocked SSH server that
-        # just prints something like: mock-ssh-server: openwisp-config restarted
-        # so you can then do:
-        #    self.assertIn('mock-ssh-server: openwisp-config restarted', sdout)
-        # if update_config() finishes successfully then status will be set to "applied"
-        self.assertEqual(device.status, 'applied')
+        device.refresh_from_db()
+        uid = self.ssh_server.users
+        shell = self.ssh_server.client(*uid)
+        shell.connect(self.ssh_server.host,
+                      port=self.ssh_server.port,
+                      username='root',
+                      pkey=dc.connector_instance.params['pkey'])
+        _, stdout, _ = shell.exec_command('echo mock-ssh-server: openwisp-config restarted')
+        while not stdout.channel.recv_ready():
+            time.sleep(0.1)
+        out = ''
+        while stdout.channel.recv_ready():
+            out += str(stdout.channel.recv(1024))
+        self.assertIn('openwisp-config restarted', out)
```

Comment:
you're executing an echo on the mock ssh server and asserting the output, for what purpose?
We need to mock the command `/etc/init.d/openwisp-config restart` so when that command is executed in the mock ssh server it print that output, so we can then capture the output and run an assertion on it.

Reply:
I myself was not convinced about this test. I was running from executing this comment on the server because from your last comment, you said this command was suppose to be executed during `config.full_clean()` and `config.save()`. 
will try to refine it. Thanks